pkg/virtctl/objectgraph/objectgraph.go

```diff
@@ -0,0 +1,131 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package objectgraph
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	"sigs.k8s.io/yaml"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/clientconfig"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+type command struct {
+	vmi            bool
+	shouldExclude  bool
+	labelSelectors map[string]string
+	outputFormat   string
+}
+
+func NewObjectGraphCommand() *cobra.Command {
+	c := command{}
+	cmd := &cobra.Command{
+		Use:     "objectgraph (VM|VMI)",
+		Short:   "Returns dependency graph related to a VM|VMI.",
+		Example: usageObjectGraph(),
+		Args:    cobra.ExactArgs(1),
+		RunE:    c.objectGraphRun,
+	}
+
+	cmd.Flags().BoolVar(&c.vmi, "vmi", false, "Returns the object graph from the VMI instead of the VM.")
+	cmd.Flags().BoolVar(&c.shouldExclude, "exclude-optional", false, "Exclude optional nodes from the object graph.")
+	cmd.Flags().StringToStringVar(&c.labelSelectors, "selector", map[string]string{}, "Label selectors to filter the object graph (multiple labels can be specified).")
+	cmd.Flags().StringVarP(&c.outputFormat, "output", "o", "json", "Output format. One of: json|yaml")
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+
+	return cmd
+}
+
+// usageObjectGraph provides several valid usage examples of objectgraph
+func usageObjectGraph() string {
+	return `
+  # Get the object graph for a VirtualMachine named 'my-vm'
+  {{ProgramName}} objectgraph my-vm
+
+  # Get the object graph for a VirtualMachineInstance named 'my-vmi'
+  {{ProgramName}} objectgraph my-vmi --vmi
```

Comment:
supersupernit - I might be wrong here but `objectgraph --vmi my-vmi` and having the arg at the end past the switch reads better to me.

Reply:
ack!